libvips/iofuncs/image.c

```diff
@@ -1689,12 +1687,14 @@ VipsImage *
 vips_image_new( void )
 {
 	VipsImage *image;
+	char filename[26];
```

Comment:
26 chars is probably enough but I wonder if there was a reason for the original 256 chars? Non-standard tmp directories?

Reply:
It might just be a leftover imperfection from 8 years ago I guess?
The old code had the same hardcoded string which can't be longer than 26 characters.

Reply:
Just habit, I think. 256 is a generic "small string" size.

The only problem perhaps is that 26 is an oddly specific value. People coming after us will wonder how it was chosen.

Reply:
I chose 26 because that's
```
strlen("temp-") + ceil(log10(1 << 63)) + /* minus sign */ 1 + /* null byte */ 1
```

Do you have a recommendation how to improve that?
Maybe using a constant in form of a macro definition?

Reply:
Maybe:

```C
#define TEMP_NAME_PREFIX "temp-"
/* Add 30 for an int in ascii, that should be enough for anyone (jk).
 */
#define TEMP_NAME_BUFFER_SIZE (sizeof (TEMP_NAME_PREFIX) + 30)

/* Make a name for a filename-less image. The buffer should be at least 
 * TEMP_NAME_BUFFER_SIZE characters. 
 */
static const char *
vips_image_temp_name( char *name, int size )
{
        static int global_serial = 0;

        int serial = g_atomic_int_add( &global_serial, 1);

        vips_snprintf( name, size, TEMP_NAME_PREFIX "-%d", serial++ );

        return( name );
}
```

I must stop, I'm not thinking clearly. I'm celebrating getting a grant of £150 (via Kirk) to organise a libvips London meeting, so I've had ahem a couple of glasses of wine. 

Anyway, this all looks great and I don't have much to add, so ... cheers!